src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,522 @@
+        True
+        sage: # needs sage.modules
+        sage: M = Matrix([(0,1,0,0,1,1,0,0,0,0),
+        ....:             (1,0,1,0,0,0,1,0,0,0),
+        ....:             (0,1,0,1,0,0,0,1,0,0),
+        ....:             (0,0,1,0,1,0,0,0,1,0),
+        ....:             (1,0,0,1,0,0,0,0,0,1),
+        ....:             (1,0,0,0,0,0,0,1,1,0),
+        ....:             (0,1,0,0,0,0,0,0,1,1),
+        ....:             (0,0,1,0,0,1,0,0,0,1),
+        ....:             (0,0,0,1,0,1,1,0,0,0),
+        ....:             (0,0,0,0,1,0,1,1,0,0)])
+        sage: M
+        [0 1 0 0 1 1 0 0 0 0]
+        [1 0 1 0 0 0 1 0 0 0]
+        [0 1 0 1 0 0 0 1 0 0]
+        [0 0 1 0 1 0 0 0 1 0]
+        [1 0 0 1 0 0 0 0 0 1]
+        [1 0 0 0 0 0 0 1 1 0]
+        [0 1 0 0 0 0 0 0 1 1]
+        [0 0 1 0 0 1 0 0 0 1]
+        [0 0 0 1 0 1 1 0 0 0]
+        [0 0 0 0 1 0 1 1 0 0]
+        sage: G = Graph(M)
+        sage: H = MatchingCoveredGraph(G)
+        sage: H == G
+        True
+        sage: # needs sage.modules
+        sage: M = Matrix([(-1, 0, 0, 0, 1, 0, 0, 0, 0, 0,-1, 0, 0, 0, 0),
+        ....:             ( 1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0, 0, 0),
+        ....:             ( 0, 1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0, 0),
+        ....:             ( 0, 0, 1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0),
+        ....:             ( 0, 0, 0, 1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1),
+        ....:             ( 0, 0, 0, 0, 0,-1, 0, 0, 0, 1, 1, 0, 0, 0, 0),
+        ....:             ( 0, 0, 0, 0, 0, 0, 0, 1,-1, 0, 0, 1, 0, 0, 0),
+        ....:             ( 0, 0, 0, 0, 0, 1,-1, 0, 0, 0, 0, 0, 1, 0, 0),
+        ....:             ( 0, 0, 0, 0, 0, 0, 0, 0, 1,-1, 0, 0, 0, 1, 0),
+        ....:             ( 0, 0, 0, 0, 0, 0, 1,-1, 0, 0, 0, 0, 0, 0, 1)])
+        sage: M
+        [-1  0  0  0  1  0  0  0  0  0 -1  0  0  0  0]
+        [ 1 -1  0  0  0  0  0  0  0  0  0 -1  0  0  0]
+        [ 0  1 -1  0  0  0  0  0  0  0  0  0 -1  0  0]
+        [ 0  0  1 -1  0  0  0  0  0  0  0  0  0 -1  0]
+        [ 0  0  0  1 -1  0  0  0  0  0  0  0  0  0 -1]
+        [ 0  0  0  0  0 -1  0  0  0  1  1  0  0  0  0]
+        [ 0  0  0  0  0  0  0  1 -1  0  0  1  0  0  0]
+        [ 0  0  0  0  0  1 -1  0  0  0  0  0  1  0  0]
+        [ 0  0  0  0  0  0  0  0  1 -1  0  0  0  1  0]
+        [ 0  0  0  0  0  0  1 -1  0  0  0  0  0  0  1]
+        sage: G = Graph(M)
+        sage: H = MatchingCoveredGraph(G)
+        sage: H == G
+        True
+        sage: G = Graph([(0, 1), (0, 3), (0, 4), (1, 2), (1, 5), (2, 3), (2, 6), (3, 7), (4, 5), (4, 7), (5, 6), (6, 7)])
+        sage: H = MatchingCoveredGraph(G)
+        sage: H == G
+        True
+        sage: import igraph                                                      # optional - python_igraph
+        sage: G = Graph(igraph.Graph([(0, 1), (0, 3), (1, 2), (2, 3)]))          # optional - python_igraph
+        sage: H = MatchingCoveredGraph(G)
+        sage: H
+        Matching covered graph on 4 vertices
```

Comment:
This test seems correct, isn't it?
The graph `H` is isomorphic to a cycle graph on 4 vertices.

Reply:
This issue was due to the absence of the the comment `# optional - python_igraph` in line no. 165 and 166. It has been resolved.